language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/ContractTypeCompanion.java

```diff
@@ -5,18 +5,22 @@
 
 import com.daml.ledger.javaapi.data.Identifier;
 
+import java.util.List;
+
 /** The commonality between {@link ContractCompanion} and {@link InterfaceCompanion}. */
 public abstract class ContractTypeCompanion<Maker, Data> {
   /** The full template ID of the template or interface that defined this companion. */
   public final Identifier TEMPLATE_ID;
+  private final List<ChoiceMetadata<Maker, ?, ?>> choices;
```

Comment:
Which can be made in the constructor here.

```suggestion
  private final Map<String, ChoiceMetadata<Marker, ?, ?>> choices;
```

Reply:
I would guess the choice name.